tests/requirements.in

```diff
@@ -2,6 +2,6 @@ pytest ~=4.3.0
 pytest-bdd ~=3.2.1
 testinfra >=2.0.0
 paramiko >=2.4.2 # testinfra backend
-cryptography >=2.4.2
+cryptography ~=3.3.2
```

Comment:
FWIW, I'm not sure we need to be as precise as this. In #3103 I changed it to:

```
cryptography ~=3.3,>=3.3.2
```

Which allows us to install any 3.x.y version, whereas your constraint forces a 3.3.z version.

Reply:
Oh? I was assuming `~=3.3.2` means 'SemVer API-compatible with 3.3.2, at least 3.3.2', so `~=3.3 && >= 3.3.2`...

Reply:
You're right, from https://www.python.org/dev/peps/pep-0440/#version-specifiers:
```
For a given release identifier V.N, the compatible release clause is approximately equivalent to the pair of comparison clauses:

>= V.N, == V.*
This operator MUST NOT be used with a single segment version number such as ~=1.

For example, the following groups of version clauses are equivalent:

~= 2.2
>= 2.2, == 2.*

~= 1.4.5
>= 1.4.5, == 1.4.*
```

Will change.